cfme/utils/appliance/__init__.py

```diff
@@ -1726,13 +1726,18 @@ def server_roles(self, roles):
         if self.server_roles == roles:
             self.log.debug(' Roles already match, returning...')
             return
-        ansible = roles.get('embedded_ansible', False)
+        ansible_old = self.server_roles('embedded_ansible', False)
+        ansible_new = roles.get('embedded_ansible', False)
+        enabling_ansible = False
+        if ansible_old is False and ansible_new is True:
```

Comment:
consider just setting `enabling_ansible` to the result of the boolean statement.

```python
enabling = (old is False and new is True)